ui/job-view/KeyboardShortcuts.jsx

```diff
@@ -0,0 +1,354 @@
+  autoclassifyChangeSelection: 'a',
+  addRelatedBug: 'b',
+  pinEditComment: 'c',
+  autoclassifyToggleEdit: 'e',
+  quickFilter: 'f',
+  clearFilter: 'ctrl+shift+f',
+  toggleInProgress: 'i',
+  autoclassifyIgnore: 'shift+i',
+  nextUnclassified: ['j', 'n'],
+  previousUnclassified: ['k', 'p'],
+  openLogviewer: 'l',
+  autoclassifySelectOption: ['o', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0'],
+  jobRetrigger: 'r',
+  autoclassifySaveAll: 's',
+  selectNextTab: 't',
+  toggleUnclassifiedFailures: 'u',
+  clearPinboard: 'ctrl+shift+u',
+  autoclassifyToggleExpandOptions: 'x',
+  previousJob: 'left',
+  nextJob: 'right',
+  previousAutoclassifyOption: ['up', 'shift+up'],
+  nextAutoclassifyOption: 'down',
+  pinJob: 'space',
+  toggleOnScreenShortcuts: '?',
+  /* these should happen regardless of being in an input field */
+  clearSelectedJob: 'escape',
+  saveClassification: 'ctrl+enter',
+  deleteClassification: 'ctrl+backspace',
+};
+
+export default class KeyboardShortcuts extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { $injector } = this.props;
+    this.$rootScope = $injector.get('$rootScope');
+  }
+
+  componentDidMount() {
+    this.autoclassifyChangeSelection = this.autoclassifyChangeSelection.bind(this);
+    this.addRelatedBug = this.addRelatedBug.bind(this);
+    this.pinEditComment = this.pinEditComment.bind(this);
+    this.autoclassifyToggleEdit = this.autoclassifyToggleEdit.bind(this);
+    this.quickFilter = this.quickFilter.bind(this);
+    this.clearFilter = this.clearFilter.bind(this);
+    this.autoclassifyIgnore = this.autoclassifyIgnore.bind(this);
+    this.nextUnclassified = this.nextUnclassified.bind(this);
+    this.previousUnclassified = this.previousUnclassified.bind(this);
+    this.openLogviewer = this.openLogviewer.bind(this);
+    this.autoclassifySelectOption = this.autoclassifySelectOption.bind(this);
+    this.jobRetrigger = this.jobRetrigger.bind(this);
+    this.autoclassifySaveAll = this.autoclassifySaveAll.bind(this);
+    this.selectNextTab = this.selectNextTab.bind(this);
+    this.clearPinboard = this.clearPinboard.bind(this);
+    this.autoclassifyToggleExpandOptions = this.autoclassifyToggleExpandOptions.bind(this);
+    this.previousJob = this.previousJob.bind(this);
+    this.nextJob = this.nextJob.bind(this);
+    this.previousAutoclassifyOption = this.previousAutoclassifyOption.bind(this);
+    this.nextAutoclassifyOption = this.nextAutoclassifyOption.bind(this);
+    this.pinJob = this.pinJob.bind(this);
+    this.clearSelectedJob = this.clearSelectedJob.bind(this);
+    this.saveClassification = this.saveClassification.bind(this);
+    this.deleteClassification = this.deleteClassification.bind(this);
+    this.toggleOnScreenShortcuts = this.toggleOnScreenShortcuts.bind(this);
+
+    // HotKeys requires focus be a component inside itself to work
+    // TODO: We may not need this if we wrap <body> with HotKeys.
+    $('#keyboard-shortcuts').focus();
```

Comment:
I don't think we need jQuery here - could we try using this?

```js
document.getElementById('keyboard-shortcuts').focus();
```

(ref http://youmightnotneedjquery.com/)

Reply:
Heh, you're totally right.  Thanks.  :)